src/main/g8/src/test/scala/$package$/ParallelismSpec.scala

```diff
@@ -10,26 +11,26 @@ class ParallelismSpec extends BaseSpec {
 
   describe("Parallelism Demo") {
 
-    implicit val s: Scheduler = Scheduler.singleThread("TestScheduler")
-
     describe("with pi calculation") {
 
+      lazy val parallelism = Runtime.getRuntime.availableProcessors()
+
       it ("should approximate pi within a digit of precision with 100 iterations") {
-        val futurePi = DemoParallelism.approximatePi(100).runToFuture
-        val pi = Await.result(futurePi, atMost = 5.seconds)
-        assert(pi > 3 && pi < 4)
+        val pio = DemoParallelism.approximatePi(parallelism, 100)
+        val timedPi = pio.timeout(5.seconds)
+        timedPi.asserting(pi => assert(pi > 3 && pi < 4))
       }
 
       it ("should approximate pi within two digits of precision with 1000 iterations") {
-        val futurePi = DemoParallelism.approximatePi(1000).runToFuture
-        val pi = Await.result(futurePi, atMost = 5.seconds)
-        assert(pi > 3.1 && pi < 3.2)
+        val pio = DemoParallelism.approximatePi(parallelism, 1000)
+        val timedPi = pio.timeout(5.seconds)
+        timedPi.asserting(pi => assert(pi > 3.1 && pi < 3.2))
       }
 
       it ("should approximate pi within three digits of precision with 10000 iterations") {
-        val futurePi = DemoParallelism.approximatePi(10000).runToFuture
-        val pi = Await.result(futurePi, atMost = 5.seconds)
-        assert(pi > 3.14 && pi < 3.16)
+        val pio = DemoParallelism.approximatePi(parallelism, 10000)
+        val timedPi = pio.timeout(5.seconds)
+        timedPi.asserting(pi => assert(pi > 3.14 && pi < 3.16))
```

Comment:
"pio", you dirty rascal.

Reply:
you have no idea how tempted I was to stick "piable" somewhere in this code base...